design-proposals/vm-backend-storage.md

```diff
@@ -0,0 +1,82 @@
+# Overview
+Today, anything that happens on the host side of VMIs (i.e. virt-launcher) is transient.  
+Any change a VMI makes outside of its own file-system is lost after a shutdown.  
+Such changes include EFI firmware settings and information stored inside the TPM.  
```

Comment:
Can you please give an example of such settings?

Reply:
Mostly the boot entries. When installing an operating system, an entry gets added to the boot manager with the EFI module to boot and its parameters.
Luckily, Windows and most Linux distros install their EFI module to a standard location, and it doesn't need any parameters, so they are still bootable after the entry is gone.
However, if a user wanted to compile their kernel directly as an EFI module and skip shim+grub, they would not be able to customize the command line without recompiling the kernel.

Other options include:
- Boot order
- System language
- Default display resolution
- Secureboot state/keys

Sorry for the delay!

Reply:
Subsequent boots are also significant faster when EFI vars are persisted.

Reply:
@rmohr do you know why boots are faster when EFI vars are persisted? Maybe thanks to the bootmgr entry that avoids browsing every EFI system partition? It still doesn't feel like that should make a perceptible difference...